app/models/skin.rb

```diff
@@ -188,6 +188,7 @@ def cache!
     self.public = true
     self.official = true
     save!
+    Rails.cache.increment('skins_generation/'+(self.type.nil? ? ("site_skin") : self.id.to_s) )
```

Comment:
This now looks right to me in terms of not over-expiring, but can I make a request? Can you move this line of code into a separate clearly named method which you call instead of the line itself wherever it appears? eg: 
```
def expire_style_block_memcache
  Rails.cache.increment('skins_generation/'+(self.type.nil? ? ("site_skin") : self.id.to_s) )
end
```

I think this will make the code more readable and also self-explanatory for what's happening, and helps to make a distinction between the file "cache" of the CSS code and the memcache of the style block.